### ISSUE TITLE
ASG group used unsorted variables map and can be overwritten if new Available zone had before sort order

### ISSUE THREAD

**Issue by akastav** [provectus/swiss-army-kube#191]
**Describe the bug**
ASG group used unsorted variables map and can be overwritten if new Available zone had before sort order.

**Use version**
All

**To Reproduce**
For example, create a cluster with 2 zones us-west-2b and us-west-2c. After that, add us-west-2a and you will get a rewrite of all ASG groups starting with us-west-2a > us-west-2b, us-west-2b > us-west-2c + us-west-2c will be created

**Expected behavior**
Saving the order of adding to the array var.availability_zones

**Comment by trolleksii** [comment]
There is a known "bug" (not a bug really, just a mechanics) in any resource created using count from the lists. Insertion of an element in the list will shift all following elements right and that cascades down the dependency graph causing recreation of entire infra if the worst case(if the item is inserted in the beginning of the list).

In this particular case it seems like sorting happens in `local.common` which has this construct:

```RB
locals {
  #...
  common = values({
    for index, az in var.availability_zones :
    az => {
      name_prefix             = "on-demand-common-${index}"
      instance_type           = var.on_demand_common_instance_type
      override_instance_types = var.on_demand_common_override_instance_types
      asg_max_size            = var.on_demand_common_max_cluster_size
      #...
```

Apparently results of function `values()` are [returned in lexicographical order by their corresponding keys, so the values will be returned in the same order as their keys would be returned from keys.](https://www.terraform.io/language/functions/values). In other words results will always be alphabetically sorted according to the AZ name.

The simplest solution would be to rewrite the locals calculation in the sak-kubernetes module and replace problematic locals block with a simple list expression:
 
```RB
locals {
#...
  common = [ for index, az in var.availability_zones :
    {
      name_prefix             = "on-demand-common-${index}"
      instance_type           = var.on_demand_common_instance_type
      override_instance_types = var.on_demand_common_override_instance_types
      asg_max_size            = var.on_demand_common_max_cluster_size
#...
```

But that would not fix it once and for all. It would work as long as AZs are added to the end of the `var.availability_zones` list; i.e. `["us-west-2b", "us-west-2c", "us-west-2a"]`. If someone will look at that variable thinking "hm... it's not alphabetically ordered, let me fix this quickly" everything will fall apart.

The only solution without such side effects is full transition to map + for_each but that would be a breaking change.

**Comment by trolleksii** [comment]
I prepared a [PR](https://github.com/provectus/sak-kubernetes/pull/5) implementing the described approach:

**Comment by trolleksii** [comment]
This issue was fixed in [PR#5](https://github.com/provectus/sak-kubernetes/pull/5).
